cabot/cabotapp/tasks.py

```diff
@@ -72,7 +72,13 @@ def run_all_checks():
     for check in checks:
         if check.should_run():
             check_queue = _classify_status_check(check.pk)
-            run_status_check.apply_async((check.pk,), queue=check_queue)
+            delay_sec = check.run_delay * 60
+            # NB: the `countdown` kwarg only works when the task is executed via the broker.
+            # When run locally (during dev / testing), it has no effect. Running tasks locally
+            # vs the broker is controlled by the `CELERY_ALWAYS_EAGER` environment variable,
+            # which must be UNSET COMPLETELY for tasks to run on the broker; setting it to
+            # False doesn't work!
+            run_status_check.apply_async((check.pk,), queue=check_queue, countdown=delay_sec)
```

Comment:
@elainearbaugh Ok I gave this some thought and came up w a couple solutions.

First, I think we should ignore run-delay for non-activity-counted checks, as it doesn't really make sense in that case. This also simplifies things.

Using your solution of `first_run` should work. My only concern is that if the check is only enabled briefly, we'll never run it. For example, say we have a check with a 5 min freq and 5 min delay, and it's enabled for 4 mins. Once we've waited out the delay, it'll have been disabled!

Anyway, here are my solutions:

**Solution 1 (using `countdown`):**

- Add `last_scheduled` time to all checks. Update when check is scheduled.
- Update check.should_run() to use last_scheduled time instead of last_run time.
- For activity-counted checks, set countdown=delay.

**Solution 2 (without `countdown`):**

- Add `last_enabled` and `last_disabled` times to the ActivityCounter model.
- When deciding if an activity-counted check should run:
  - Compute a future time window in which checks may run:
    - `window_start = last_enabled + run_delay`
    - `window_end = last_disabled + run_delay`
  - A check may run if now() falls within the future window and the check hasn't run recently:
    - `now() > window_start` and `now() > last_run + freq` and `now() < window_end`
    - Accounting of course for any null values, or `last_disabled < last_enabled`

Solution 1 is a bit simpler, but solution 2 has benefits:

- We avoid the use of `countdown`. I know you have concerns about it.  TBH I'm wary of it as well as I don't know if it'll behave without problems.
- The `last_enabled` and `last_disabled` fields will come in handy for visibility and debugging, i.e. to see exactly when a check was enabled/disabled.  I could already use this to debug bank pipeline checks.

Thoughts?

Reply:
I think both work! To avoid `countdown`, I'd also prefer 2), plus more visibility is always good! 

Are there any concerns that checks run after the check is supposed to be disabled may have no data? If not approach 2 sounds good to me, thanks!